cmake/BLTInstallableMacros.cmake

```diff
@@ -428,37 +428,79 @@ endmacro(blt_add_target_compile_flags)
 
 
 ##------------------------------------------------------------------------------
-## blt_convert_to_system_includes(TARGET <target>)
+## blt_convert_to_system_includes(TARGETS [<target>...]
+##                                CHILDREN [TRUE|FALSE]
+##                                [QUIET])
 ##
 ## Converts existing interface includes to system interface includes.
 ##------------------------------------------------------------------------------
-macro(blt_convert_to_system_includes)
-    set(options)
-    set(singleValuedArgs TARGET)
-    set(multiValuedArgs)
+function(blt_convert_to_system_includes)
+    set(options QUIET)
+    set(singleValuedArgs CHILDREN TARGET)
+    set(multiValuedArgs TARGETS)
 
     ## parse the arguments to the macro
     cmake_parse_arguments(arg
-         "${options}" "${singleValuedArgs}" "${multiValuedArgs}" ${ARGN})
+        "${options}" "${singleValuedArgs}" "${multiValuedArgs}" ${ARGN})
+
+    if(DEFINED arg_TARGET)
+        message(WARNING "TARGET is a deprecated parameter for the blt_convert_to_system_includes macro.")
+        list(APPEND arg_TARGETS ${arg_TARGET})
+    endif()
+
+    if(NOT DEFINED arg_TARGETS)
+        message(FATAL_ERROR "TARGETS is a required parameter for the blt_convert_to_system_includes macro.")
+    endif()
 
-    if(NOT DEFINED arg_TARGET)
-       message(FATAL_ERROR "TARGET is a required parameter for the blt_convert_to_system_includes macro.")
+    if(NOT DEFINED arg_CHILDREN)
+        set(arg_CHILDREN TRUE)
     endif()
 
     # PGI does not support -isystem
     if(NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "PGI")
-        get_target_property(_include_dirs ${arg_TARGET} INTERFACE_INCLUDE_DIRECTORIES)
-        # Don't copy if the target had no include directories
-        if(_include_dirs)
-            # Clear previous value in INTERFACE_INCLUDE_DIRECTORIES so it is not doubled
-            # by target_include_directories
-            set_property(TARGET ${arg_TARGET} PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
-            target_include_directories(${arg_TARGET} SYSTEM INTERFACE ${_include_dirs})
-        endif()
-    endif()
+        set(_target_list)
+
+        foreach(_target ${arg_TARGETS})
+            if(TARGET ${_target})
+                if(${arg_CHILDREN})
+                    blt_find_target_dependencies(TARGET ${_target} TLIST _target_list)
```

Comment:
Is it safe to call macros from BLTPrivateMacros.cmake in BLTInstallableMacros.cmake?

Reply:
no.. but you can move that macro into the installable file. I could actually use it in Axom's installed config.

Reply:
I moved it there. Should I add sphinx documentation for that macro now, too? Or will we still consider it to be a private macro?

Reply:
Just documented it.